pinterest/teletraan
# Deploy board and deploy service changes for auto decrement capacity

# Changes
- Introduced unified the experience for terminate and force terminate host(s). 
- Added support for the `replaceHost` flag in deploy board and deploy service. 
- Introduced a new state `PENDING_TERMINATE_NO_REPLACE` to `HostState` and updated all references to `PENDING_TERMINATE`, including DB query templates.
    - Note that the existing `PENDING_TERMINATE` state corresponds to the "replace host" behavior. This is intentional because the "replace host" behavior is the default and also during roll out we will not accidentally decrement desired capacities for clusters.  

## Screenshots
### Individual termination
![individual terminate](https://user-images.githubusercontent.com/8442875/189232854-8902d571-0b61-4cdf-9a0d-efb026e32e1b.png)

### Force termination
![force terminate](https://user-images.githubusercontent.com/8442875/189232939-20727572-e448-4543-986c-d1dd1487b4be.png)


### Batch termination
<img width="1652" alt="batch termination" src="https://user-images.githubusercontent.com/8442875/189200486-8b578eb2-046a-42e4-9f3c-ae1b4f214c85.png">

### Force Batch Termination
<img width="1652" alt="batch force termination" src="https://user-images.githubusercontent.com/8442875/189200501-0d98ed39-ec23-4ebb-94a1-fb31bf0f6cc5.png">

# Test plan
1. Visit host detail page, check both _terminate host_ and _force terminate_ work.
    1. Host is terminated.
    2. Host is replaced when replace host is specified.
    3. Capacity is decreased when replace host is not specified.
4. Visit all hosts view, check both batch terminate and batch force terminate work.  
    1. Host is terminated.
    2. Host is replaced when replace host is specified.
    3. Capacity is decreased when replace host is not specified.
